File: back_old/src/main/kotlin/ru/lanit/rnd/files/service/EventGenerator.java
package ru.lanit.rnd.files.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;
import ru.lanit.rnd.files.model.Event;

import java.util.concurrent.atomic.AtomicInteger;

@Component
public class EventGenerator {

    private AtomicInteger counter = new AtomicInteger(0);

    private EventUnicastService eventUnicastService;

    @Autowired
    public EventGenerator(EventUnicastService eventUnicastService) {
        this.eventUnicastService = eventUnicastService;
    }

    @Scheduled(initialDelay = 1000, fixedDelay = 1000)
    public void generateEvent() {
        int count = counter.getAndIncrement();
        Event event = new Event("event", count);
        eventUnicastService.onNext(event);
    }
}

File: back/src/main/kotlin/ru/lanit/rnd/files/service/EventGenerator.kt
package ru.lanit.rnd.files.service;

import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component
import ru.lanit.rnd.files.model.Event
import ru.lanit.rnd.files.repository.FileRepository
import java.util.concurrent.atomic.AtomicInteger

@Component
class EventGenerator(
    private val eventUnicastService: EventUnicastService,
    private val fileRepository: FileRepository
) {

    private val counter: AtomicInteger = AtomicInteger(0);

//    @Scheduled(initialDelay = 10000, fixedDelay = 10000)
//    public fun generateEvent() {
//        fileRepository.findAll().subscribe {
//            val count: Int = counter.getAndIncrement();
//            val event: Event = Event("add", it);
//            eventUnicastService.onNext(event);
//        }
//
//    }
}
